src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.cs

```diff
@@ -184,6 +184,13 @@ internal static BoundBlock ConstructAutoPropertyAccessorBody(SourceMemberMethodS
             }
 
             var field = property.BackingField;
+            if (field is null)
+            {
+                // This happens for public int { set; } where we produce ERR_AutoPropertyMustHaveGetAccessor
+                Debug.Assert(!property.HasGetAccessor);
+                return null;
```

Comment:
>return null; [](http://example.com/codeflow?start=16&length=12)

Consider adjusting the doc comment to indicate that null can be returned in case of an error.